hathor/consensus/transaction_consensus.py

```diff
@@ -296,7 +296,7 @@ def check_conflicts(self, tx: Transaction) -> None:
         for candidate in candidates:
             tx_meta = candidate.get_metadata()
             if not tx_meta.voided_by:
-                candidate.update_accumulated_weight(stop_value=meta.accumulated_weight)
+                candidate.update_accumulated_weight(stop_value=work_to_weight(meta.accumulated_weight))
```

Comment:
We were passing an `int` to a method that takes `float`, why didn't mypy catch it??

Reply:
Apparently it's not an error, maybe because integers coerce to float, I tried this and it passes:

```
def foo(i: float) -> float:
    return i * 0.99

foo(1)
```

So I think we would be better off having `Weight` and `Work` new-types.

Reply:
Right, https://mypy.readthedocs.io/en/latest/duck_type_compatibility.html